### DISCUSSION TITLE
AMP support?

### DISCUSSION THREAD

**Discussion by sudokpr** [timlrx/tailwind-nextjs-starter-blog#362] [General]
First of all, thanks a ton for the "tailwind-nextjs-starter-blog" template. It really helps speed up setting a new, professional-style blog.
I am trying a food blog based on this repo and was trying to experiment if I can add AMP support.
I am facing some challenges due to the custom NextJS App component in `_app.js`.
I am not able to conditionally skip the initialization based on `useAmp()`.
The `useAmp()` always returns false at the `_app.js`. This could be due to the way NextJS is designed to overcome to React Hydrate mismatch issue. https://github.com/vercel/next.js/issues/8259

Is there any other way to support AMP pages by conditionally rendering two different formats of the same NextJS page?

**Comment by timlrx** [comment]
Hi, I don't have any plans on supporting AMP. By most accounts, it seems to be phasing out. 

I am guessing you are already included `export const config = { amp: 'hybrid' }`? If so, based on the documentation the hook will return `true` only for `path?amp=1`. If it returns `false`, I am not sure what's the issue and will leave it for the rest who are more knowledgeable to chime in.

**Reply by sudokpr** [reply]
Yes, I did try the `path?amp=1` in the URL. It works on all the pages except for the `_app.js` and `_document.js`.
My main intention was to skip adding the header components, CSS loading etc., which are defined on the `_app.js` and `_document.js`

Thanks for your input.
This question is a more general to NextJS and not specific to this repo. I will move this discussion to NextJS forums.
